micrometer-test/src/main/java/io/micrometer/core/instrument/HttpClientTimingInstrumentationVerificationSuite.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test suite for HTTP client timing instrumentation that verifies the expected metrics
+ * are registered and recorded after different scenarios. Use this suite to ensure that
+ * your instrumentation has the expected naming and tags. A locally running server is used
+ * to receive real requests from an instrumented HTTP client.
+ */
+@WireMockTest
+public abstract class HttpClientTimingInstrumentationVerificationSuite extends InstrumentationVerificationSuite {
+
+    enum HttpMethod {
+
+        GET, POST;
+
+    }
+
+    /**
+     * A default is provided that should be preferred by new instrumentations. Existing
+     * instrumentations that use a different value to maintain backwards compatibility may
+     * override this method to run tests with a different name used in assertions.
+     * @return name of the meter timing http client requests
+     */
+    protected String timerName() {
+        return "http.client.requests";
```

Comment:
A future enhancement idea: We could inject the new naming convention mechanism here - that way if someone wants to see if they are compatible with a given standard they will be able to easily do it